main.cpp

```diff
@@ -20,19 +21,39 @@ struct Wall
 {
     plane p;
     color c;
+
+    friend std::ostream& operator<<(std::ostream& os, const Wall&wall) {
```

Comment:
I don't understand what's the point of a friend member here. All of the `struct` members are public by default and visible to everybody. We are not doing OOP for this particular project, so encapsulation is not needed. I want structs to be just [Composite data types](https://en.wikipedia.org/wiki/Composite_data_type) without any member functions.

Please remove the `friend` keyword and move all of the `operator<<` outside of the structs.